src/main/python/ayab/statusbar.py

```diff
@@ -18,15 +18,19 @@
 #    https://github.com/AllYarnsAreBeautiful/ayab-desktop
 """Notification methods using the status bar."""
 
+from __future__ import annotations
 import logging
 from PySide6.QtWidgets import QStatusBar
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .ayab import GuiMain
 
 
 class StatusBar(QStatusBar):
-    def __init__(self, parent):
+    def __init__(self, parent:GuiMain):
         super().__init__(parent)
 
-    def update(self, text, log=True):
+    def update(self, text:str, log:bool=True)->None: #type: ignore
```

Comment:
this overrides the supertype incorrectly, but I don't think it matters enough to fight over, so we're ignoring the type for now.